apps/dashr-tsne/app.R

```diff
@@ -0,0 +1,1195 @@
+    
+    return(p)
+  }
+  # Nearest Neighbors display only the 100 nearest neighbors of the selected_word, in text rather than circles
+  else if(wordemb_display_mode == 'neighbors'){
+    if(selected_word == ''){
+      return(plot_ly(type =  'scatter3d', x=0, y=0, z=0, mode='text', text='NA'))
+    }
+    plot_mode = 'text'
+    #Get the nearest neighbors incides using Euclidean distance
+    
+    
+    
+    
+    selected_vector = filter(datTbl, label==selected_word)
+    mtx <- as.matrix(datTbl[, -1])
+    sel_vec <- unlist(selected_vector[, -1])
+    
+    distances <- mtx %>% apply(., 1, . %>% eu.norm(., sel_vec))
+    datTbl$distance = distances
+    sorted_DT <- datTbl[order(distance), ]
+    neighbors_label <- sorted_DT[1:101, 1]
+    neighbors <- embedding_df[label %in% unlist(neighbors_label)]
+    return(plot_ly(neighbors, type='scatter3d', x=~x, y=~y, z=~z ,name=~label,text=~label, textposition='middle-center', showlegend=FALSE, mode='text'))
+    
+  }
+}
+
+app$callback(
+  output = l.(id='tsne-3d-plot', property='figure')
+  ,
+  params = l.(input(id='dropdown-dataset', property='value')
+              ,
+              input(id='slider-iterations',  property='value')
+              ,
+              input(id='slider-perplexity',  property='value')
+              ,
+              input(id='slider-pca-dimensions',  property='value')
+              ,
+              input(id='slider-learning-rate',  property='value')
+              ,
+              input(id='dropdown-word-selected',  property='value')
+              ,
+              input(id='radio-wordemb-display-mode',  property='value')
+              
+  )
+  ,
+  
+  
+  function(dataset, iterations, perplexity, pca_dim, learning_rate, selected_word, wordemb_display_mode){
+    if(dataset==''|| is.null(dataset)){
+      return(defaultPlot)
+    } else {
+      axes <- l.(title='', showgrid=T, zeroline=F, showticklabels=F)
+      path = as.character(glue('demo_embeddings/{dataset}/iterations_{iterations}/perplexity_{perplexity}/pca_{pca_dim}/learning_rate_{learning_rate}/data.csv'))
+      embedding_DT <- fread(path, encoding='Latin-1', header=T)
+      datTbl <- datLst[[dataset]]
+      colnames(datTbl)[1] <- 'label'
+      if(is.element(dataset, IMAGE_DATASETS)){
+        
+        colnames(embedding_DT) <- c('label', 'x','y','z')
+        p <- plot_ly(embedding_DT, type='scatter3d', x=~x, y=~y, z=~z, color=~as.factor(label), mode='markers', marker=list(symbol='circle', size=2.5))
+        
+        return(p)
+      } else if(is.element(dataset, WORD_EMBEDDINGS)){
+        figure <- generate_figure_word_vec(embedding_DT, wordemb_display_mode, selected_word, datTbl)
+        
+        return(figure)
+      }else{
+        return(plot_ly(type =  'scatter3d'))}
+      
+    }
+  }
+)
+#
+#### Returning the 5 nearest neighborhood for the point clicked
+
+app$callback(
```

Comment:
The bar plot seems to spill over the div:
![2019-06-20_17:26:11](https://user-images.githubusercontent.com/19624164/59882566-8bad5400-9380-11e9-9ca0-4c1d2ed2e548.png)

Reply:
This issue was the one that got me stuck over a while - now it has been solved!